flutter/ios/Classes/SwiftSentryFlutterPlugin.swift

```diff
@@ -3,27 +3,216 @@ import Sentry
+
+        if (arguments.isEmpty) {
+            result(FlutterError(code: "4", message: "Arguments is null or empty", details: nil) )
+            return
+        }
+
+        SentrySDK.start { options in
+            if(arguments["dsn"] != nil ){
+                options.dsn = arguments["dsn"] as? String
+            } else {
+                result(FlutterError(code: "4", message: "A valid Dsn must be provided", details: nil) )
+            }
+
+            if let isDebug = arguments["debug"] as? Bool {
+                options.debug = isDebug
+            }
+
+            if let environment = arguments["environment"] as? String{
+                options.environment = environment
+            }
+
+            if let releaseName = arguments["release"] as? String{
+                options.releaseName = releaseName
+            }
+
+            if let enableAutoSessionTracking = arguments["enableAutoSessionTracking"] as? Bool {
+                options.enableAutoSessionTracking = enableAutoSessionTracking
+            }
+
+            if let attachStacktrace = arguments["attachStacktrace"] as? Bool {
+                options.attachStacktrace = attachStacktrace
+            }
+
+            if let diagnosticLevel = arguments["diagnosticLevel"] as? String, options.debug == true {
+
+                switch (diagnosticLevel) {
+                case "fatal", "error":
+                    options.logLevel = .error
+                    break;
+                case "debug":
+                    options.logLevel = .debug
+                    break;
+                case "warning", "info":
+                    options.logLevel = .verbose
+                    break;
+                default:
+                    options.logLevel = .none
+                    break;
+                }
+            }
+
+            if let sessionTrackingIntervalMillis = arguments["sessionTrackingIntervalMillis"] as? UInt{
+                options.sessionTrackingIntervalMillis = sessionTrackingIntervalMillis
+            }
+
+            if let dist = arguments["dist"] as? String{
+                options.dist = dist
+            }
+
+            if let integrations = arguments["integrations"] as? [String]{
+                options.integrations?.append(contentsOf: integrations)
+            }
+
+            if let enableAutoNativeBreadcrumbs = arguments["enableAutoNativeBreadcrumbs"] as? Bool,
+               enableAutoNativeBreadcrumbs == false {
+                options.integrations = options.integrations?.filter { (name) -> Bool in
+                    return name != "SentryAutoBreadcrumbTrackingIntegration"
+                }
+            }
+
+            if let maxBreadcrumbs = arguments["maxBreadcrumbs"] as? UInt{
+                options.maxBreadcrumbs = maxBreadcrumbs
+            }
+
+            /*
+             missing fields : enableNativeCrashHandling, packages,anrTimeoutIntervalMillis, cacheDirSize
+             */
+
+            options.beforeSend = { event in
```

Comment:
beforeSend seems to not be called for now with `captureEnvelope` https://github.com/getsentry/sentry-cocoa/blob/c34b9067584b52e4136e5fddb7050ee6f3ca022a/Sources/Sentry/SentryClient.m#L243-L253

Reply:
in this case, we'd need to merge this into the event before calling `captureEnvelope()`, so the envelope contains this data, I'm confirming this, hold on.